core/domain/interaction_jobs_one_off.py

```diff
@@ -392,3 +393,48 @@ def get_invalid_values(value_type, value, choices):
     @staticmethod
     def reduce(key, values):
         yield (key, values)
+
+
+class LogicProofInteractionOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that produces a list of (exploaration, state) pairs that use the
+    logicProof interaction. This will be used to send manual emails.
+    This job can be deleted after logicProof is removed from the codebase.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        user_email = None
+        for state in exploration.states.values():
+            if state.interaction.id == 'LogicProof':
+                exploration_rights = rights_manager.get_exploration_rights(
+                    item.id, strict=False)
+                if exploration_rights is None:
+                    yield ('MISSING_RIGHTS', item.id)
+                    yield ('SUCCESS', 1)
+                    return
+
+                exp_owner_ids = exploration_rights.owner_ids
+                if len(exp_owner_ids) == 0:
+                    yield ('EMPTY', item.id)
+                    yield ('SUCCESS', 1)
+                    return
+                for user_id in exp_owner_ids:
+                    user_email = user_services.get_email_from_user_id(user_id)
+                    if user_email:
+                        yield ('EMAIL_DATA', (user_email, item.id))
```

Comment:
Again this would mean that we would sometimes print duplicity emails if exploration had multiple states with LogicProof interaction. You should do
```suggestion
                for user_id in exp_owner_ids:
                    user_email = user_services.get_email_from_user_id(user_id)
                    if user_email:
                        yield ('EMAIL_DATA', (user_email, item.id))
                yield ('SUCCESS', 1)
                return
```

Reply:
okay I had a misunderstanding. Thanks its updated, PTAL!